tests/cobs_test.cc

```diff
@@ -0,0 +1,145 @@
+#include "../inc/cobs.h"
+
+/*
+Throw exceptions so we can catch them with gdb (catch catch).
+*/
+void throw_assert(bool condition, const char * const remark) {
+  if (condition == false) throw remark;
```

Comment:
I've found it much more powerful and less tedious to simply debug your tests than to try and give all information you could possibly need to debug the problem in the form of an error message + stack trace. 

This is a simple solution to simple problem. I can spend time on adopting a testing framework but that is not a priority right?